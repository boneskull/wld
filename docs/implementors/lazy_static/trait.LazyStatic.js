(function() {var implementors = {};
implementors["wld"] = [{"text":"impl LazyStatic for BLOCKTYPE_COLOR_MAP","synthetic":false,"types":[]},{"text":"impl LazyStatic for WALLTYPE_COLOR_MAP","synthetic":false,"types":[]},{"text":"impl LazyStatic for PAINT_COLOR_MAP","synthetic":false,"types":[]}];
if (window.register_implementors) {window.register_implementors(implementors);} else {window.pending_implementors = implementors;}})()