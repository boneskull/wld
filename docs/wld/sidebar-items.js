initSidebarItems({"fn":[["enable_logger","Enable logging for this crate given [`log::LevelFilter`] `level`."],["parse_world","Instantiate a [`World`] from a slice of bytes."]],"mod":[["constants",""],["enums",""],["models",""]]});