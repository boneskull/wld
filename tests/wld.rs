use insta::assert_debug_snapshot;
use lazy_static::lazy_static;
use std::fs::read;
use wld::{model::world::World, parse_world};

lazy_static! {
  // paths are relative to root, I guess?
  pub static ref WORLD: std::vec::Vec<u8> =
    read("tests/fixtures/Foon.wld").expect("Unable to read file");
  pub static ref PARSED_WORLD: World = parse_world(&WORLD).unwrap();
}

#[test]
fn test_parse_header() {
  assert_debug_snapshot!(PARSED_WORLD.header);
}

#[test]
fn test_parse_properties() {
  assert_debug_snapshot!(PARSED_WORLD.properties);
}

#[test]
fn test_parse_status() {
  assert_debug_snapshot!(PARSED_WORLD.status);
}
