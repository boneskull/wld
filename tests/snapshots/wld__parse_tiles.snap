---
source: tests/wld.rs
expression: "&PARSED_WORLD.tiles[0][0..50]"
---
[
    Tile {
        tile_header: TileHeader {
            has_block: false,
            has_attributes: false,
            has_wall: false,
            liquid_type: NoLiquid,
            has_extended_block_id: false,
            rle_type: SingleByte,
            bits: TBitVec(
                BitVec<Lsb0, u8> [
                    0b00000010,
                ],
            ),
        },
        attributes: None,
        ext_attributes: None,
        block: None,
        wall: None,
        liquid: None,
        wiring: None,
        run_length: RunLength {
            length: 98,
            rle_type: SingleByte,
        },
        position: Position {
            x: 0,
            y: 0,
        },
    },
    Tile {
        tile_header: TileHeader {
            has_block: false,
            has_attributes: false,
            has_wall: false,
            liquid_type: NoLiquid,
            has_extended_block_id: false,
            rle_type: SingleByte,
            bits: TBitVec(
                BitVec<Lsb0, u8> [
                    0b00000010,
                ],
            ),
        },
        attributes: None,
        ext_attributes: None,
        block: None,
        wall: None,
        liquid: None,
        wiring: None,
        run_length: RunLength {
            length: 98,
            rle_type: SingleByte,
        },
        position: Position {
            x: 0,
            y: 0,
        },
    },
    Tile {
        tile_header: TileHeader {
            has_block: false,
            has_attributes: false,
            has_wall: false,
            liquid_type: NoLiquid,
            has_extended_block_id: false,
            rle_type: SingleByte,
            bits: TBitVec(
                BitVec<Lsb0, u8> [
                    0b00000010,
                ],
            ),
        },
        attributes: None,
        ext_attributes: None,
        block: None,
        wall: None,
        liquid: None,
        wiring: None,
        run_length: RunLength {
            length: 98,
            rle_type: SingleByte,
        },
        position: Position {
            x: 0,
            y: 0,
        },
    },
    Tile {
        tile_header: TileHeader {
            has_block: false,
            has_attributes: false,
            has_wall: false,
            liquid_type: NoLiquid,
            has_extended_block_id: false,
            rle_type: SingleByte,
            bits: TBitVec(
                BitVec<Lsb0, u8> [
                    0b00000010,
                ],
            ),
        },
        attributes: None,
        ext_attributes: None,
        block: None,
        wall: None,
        liquid: None,
        wiring: None,
        run_length: RunLength {
            length: 98,
            rle_type: SingleByte,
        },
        position: Position {
            x: 0,
            y: 0,
        },
    },
    Tile {
        tile_header: TileHeader {
            has_block: false,
            has_attributes: false,
            has_wall: false,
            liquid_type: NoLiquid,
            has_extended_block_id: false,
            rle_type: SingleByte,
            bits: TBitVec(
                BitVec<Lsb0, u8> [
                    0b00000010,
                ],
            ),
        },
        attributes: None,
        ext_attributes: None,
        block: None,
        wall: None,
        liquid: None,
        wiring: None,
        run_length: RunLength {
            length: 98,
            rle_type: SingleByte,
        },
        position: Position {
            x: 0,
            y: 0,
        },
    },
    Tile {
        tile_header: TileHeader {
            has_block: false,
            has_attributes: false,
            has_wall: false,
            liquid_type: NoLiquid,
            has_extended_block_id: false,
            rle_type: SingleByte,
            bits: TBitVec(
                BitVec<Lsb0, u8> [
                    0b00000010,
                ],
            ),
        },
        attributes: None,
        ext_attributes: None,
        block: None,
        wall: None,
        liquid: None,
        wiring: None,
        run_length: RunLength {
            length: 98,
            rle_type: SingleByte,
        },
        position: Position {
            x: 0,
            y: 0,
        },
    },
    Tile {
        tile_header: TileHeader {
            has_block: false,
            has_attributes: false,
            has_wall: false,
            liquid_type: NoLiquid,
            has_extended_block_id: false,
            rle_type: SingleByte,
            bits: TBitVec(
                BitVec<Lsb0, u8> [
                    0b00000010,
                ],
            ),
        },
        attributes: None,
        ext_attributes: None,
        block: None,
        wall: None,
        liquid: None,
        wiring: None,
        run_length: RunLength {
            length: 98,
            rle_type: SingleByte,
        },
        position: Position {
            x: 0,
            y: 0,
        },
    },
    Tile {
        tile_header: TileHeader {
            has_block: false,
            has_attributes: false,
            has_wall: false,
            liquid_type: NoLiquid,
            has_extended_block_id: false,
            rle_type: SingleByte,
            bits: TBitVec(
                BitVec<Lsb0, u8> [
                    0b00000010,
                ],
            ),
        },
        attributes: None,
        ext_attributes: None,
        block: None,
        wall: None,
        liquid: None,
        wiring: None,
        run_length: RunLength {
            length: 98,
            rle_type: SingleByte,
        },
        position: Position {
            x: 0,
            y: 0,
        },
    },
    Tile {
        tile_header: TileHeader {
            has_block: false,
            has_attributes: false,
            has_wall: false,
            liquid_type: NoLiquid,
            has_extended_block_id: false,
            rle_type: SingleByte,
            bits: TBitVec(
                BitVec<Lsb0, u8> [
                    0b00000010,
                ],
            ),
        },
        attributes: None,
        ext_attributes: None,
        block: None,
        wall: None,
        liquid: None,
        wiring: None,
        run_length: RunLength {
            length: 98,
            rle_type: SingleByte,
        },
        position: Position {
            x: 0,
            y: 0,
        },
    },
    Tile {
        tile_header: TileHeader {
            has_block: false,
            has_attributes: false,
            has_wall: false,
            liquid_type: NoLiquid,
            has_extended_block_id: false,
            rle_type: SingleByte,
            bits: TBitVec(
                BitVec<Lsb0, u8> [
                    0b00000010,
                ],
            ),
        },
        attributes: None,
        ext_attributes: None,
        block: None,
        wall: None,
        liquid: None,
        wiring: None,
        run_length: RunLength {
            length: 98,
            rle_type: SingleByte,
        },
        position: Position {
            x: 0,
            y: 0,
        },
    },
    Tile {
        tile_header: TileHeader {
            has_block: false,
            has_attributes: false,
            has_wall: false,
            liquid_type: NoLiquid,
            has_extended_block_id: false,
            rle_type: SingleByte,
            bits: TBitVec(
                BitVec<Lsb0, u8> [
                    0b00000010,
                ],
            ),
        },
        attributes: None,
        ext_attributes: None,
        block: None,
        wall: None,
        liquid: None,
        wiring: None,
        run_length: RunLength {
            length: 98,
            rle_type: SingleByte,
        },
        position: Position {
            x: 0,
            y: 0,
        },
    },
    Tile {
        tile_header: TileHeader {
            has_block: false,
            has_attributes: false,
            has_wall: false,
            liquid_type: NoLiquid,
            has_extended_block_id: false,
            rle_type: SingleByte,
            bits: TBitVec(
                BitVec<Lsb0, u8> [
                    0b00000010,
                ],
            ),
        },
        attributes: None,
        ext_attributes: None,
        block: None,
        wall: None,
        liquid: None,
        wiring: None,
        run_length: RunLength {
            length: 98,
            rle_type: SingleByte,
        },
        position: Position {
            x: 0,
            y: 0,
        },
    },
    Tile {
        tile_header: TileHeader {
            has_block: false,
            has_attributes: false,
            has_wall: false,
            liquid_type: NoLiquid,
            has_extended_block_id: false,
            rle_type: SingleByte,
            bits: TBitVec(
                BitVec<Lsb0, u8> [
                    0b00000010,
                ],
            ),
        },
        attributes: None,
        ext_attributes: None,
        block: None,
        wall: None,
        liquid: None,
        wiring: None,
        run_length: RunLength {
            length: 98,
            rle_type: SingleByte,
        },
        position: Position {
            x: 0,
            y: 0,
        },
    },
    Tile {
        tile_header: TileHeader {
            has_block: false,
            has_attributes: false,
            has_wall: false,
            liquid_type: NoLiquid,
            has_extended_block_id: false,
            rle_type: SingleByte,
            bits: TBitVec(
                BitVec<Lsb0, u8> [
                    0b00000010,
                ],
            ),
        },
        attributes: None,
        ext_attributes: None,
        block: None,
        wall: None,
        liquid: None,
        wiring: None,
        run_length: RunLength {
            length: 98,
            rle_type: SingleByte,
        },
        position: Position {
            x: 0,
            y: 0,
        },
    },
    Tile {
        tile_header: TileHeader {
            has_block: false,
            has_attributes: false,
            has_wall: false,
            liquid_type: NoLiquid,
            has_extended_block_id: false,
            rle_type: SingleByte,
            bits: TBitVec(
                BitVec<Lsb0, u8> [
                    0b00000010,
                ],
            ),
        },
        attributes: None,
        ext_attributes: None,
        block: None,
        wall: None,
        liquid: None,
        wiring: None,
        run_length: RunLength {
            length: 98,
            rle_type: SingleByte,
        },
        position: Position {
            x: 0,
            y: 0,
        },
    },
    Tile {
        tile_header: TileHeader {
            has_block: false,
            has_attributes: false,
            has_wall: false,
            liquid_type: NoLiquid,
            has_extended_block_id: false,
            rle_type: SingleByte,
            bits: TBitVec(
                BitVec<Lsb0, u8> [
                    0b00000010,
                ],
            ),
        },
        attributes: None,
        ext_attributes: None,
        block: None,
        wall: None,
        liquid: None,
        wiring: None,
        run_length: RunLength {
            length: 98,
            rle_type: SingleByte,
        },
        position: Position {
            x: 0,
            y: 0,
        },
    },
    Tile {
        tile_header: TileHeader {
            has_block: false,
            has_attributes: false,
            has_wall: false,
            liquid_type: NoLiquid,
            has_extended_block_id: false,
            rle_type: SingleByte,
            bits: TBitVec(
                BitVec<Lsb0, u8> [
                    0b00000010,
                ],
            ),
        },
        attributes: None,
        ext_attributes: None,
        block: None,
        wall: None,
        liquid: None,
        wiring: None,
        run_length: RunLength {
            length: 98,
            rle_type: SingleByte,
        },
        position: Position {
            x: 0,
            y: 0,
        },
    },
    Tile {
        tile_header: TileHeader {
            has_block: false,
            has_attributes: false,
            has_wall: false,
            liquid_type: NoLiquid,
            has_extended_block_id: false,
            rle_type: SingleByte,
            bits: TBitVec(
                BitVec<Lsb0, u8> [
                    0b00000010,
                ],
            ),
        },
        attributes: None,
        ext_attributes: None,
        block: None,
        wall: None,
        liquid: None,
        wiring: None,
        run_length: RunLength {
            length: 98,
            rle_type: SingleByte,
        },
        position: Position {
            x: 0,
            y: 0,
        },
    },
    Tile {
        tile_header: TileHeader {
            has_block: false,
            has_attributes: false,
            has_wall: false,
            liquid_type: NoLiquid,
            has_extended_block_id: false,
            rle_type: SingleByte,
            bits: TBitVec(
                BitVec<Lsb0, u8> [
                    0b00000010,
                ],
            ),
        },
        attributes: None,
        ext_attributes: None,
        block: None,
        wall: None,
        liquid: None,
        wiring: None,
        run_length: RunLength {
            length: 98,
            rle_type: SingleByte,
        },
        position: Position {
            x: 0,
            y: 0,
        },
    },
    Tile {
        tile_header: TileHeader {
            has_block: false,
            has_attributes: false,
            has_wall: false,
            liquid_type: NoLiquid,
            has_extended_block_id: false,
            rle_type: SingleByte,
            bits: TBitVec(
                BitVec<Lsb0, u8> [
                    0b00000010,
                ],
            ),
        },
        attributes: None,
        ext_attributes: None,
        block: None,
        wall: None,
        liquid: None,
        wiring: None,
        run_length: RunLength {
            length: 98,
            rle_type: SingleByte,
        },
        position: Position {
            x: 0,
            y: 0,
        },
    },
    Tile {
        tile_header: TileHeader {
            has_block: false,
            has_attributes: false,
            has_wall: false,
            liquid_type: NoLiquid,
            has_extended_block_id: false,
            rle_type: SingleByte,
            bits: TBitVec(
                BitVec<Lsb0, u8> [
                    0b00000010,
                ],
            ),
        },
        attributes: None,
        ext_attributes: None,
        block: None,
        wall: None,
        liquid: None,
        wiring: None,
        run_length: RunLength {
            length: 98,
            rle_type: SingleByte,
        },
        position: Position {
            x: 0,
            y: 0,
        },
    },
    Tile {
        tile_header: TileHeader {
            has_block: false,
            has_attributes: false,
            has_wall: false,
            liquid_type: NoLiquid,
            has_extended_block_id: false,
            rle_type: SingleByte,
            bits: TBitVec(
                BitVec<Lsb0, u8> [
                    0b00000010,
                ],
            ),
        },
        attributes: None,
        ext_attributes: None,
        block: None,
        wall: None,
        liquid: None,
        wiring: None,
        run_length: RunLength {
            length: 98,
            rle_type: SingleByte,
        },
        position: Position {
            x: 0,
            y: 0,
        },
    },
    Tile {
        tile_header: TileHeader {
            has_block: false,
            has_attributes: false,
            has_wall: false,
            liquid_type: NoLiquid,
            has_extended_block_id: false,
            rle_type: SingleByte,
            bits: TBitVec(
                BitVec<Lsb0, u8> [
                    0b00000010,
                ],
            ),
        },
        attributes: None,
        ext_attributes: None,
        block: None,
        wall: None,
        liquid: None,
        wiring: None,
        run_length: RunLength {
            length: 98,
            rle_type: SingleByte,
        },
        position: Position {
            x: 0,
            y: 0,
        },
    },
    Tile {
        tile_header: TileHeader {
            has_block: false,
            has_attributes: false,
            has_wall: false,
            liquid_type: NoLiquid,
            has_extended_block_id: false,
            rle_type: SingleByte,
            bits: TBitVec(
                BitVec<Lsb0, u8> [
                    0b00000010,
                ],
            ),
        },
        attributes: None,
        ext_attributes: None,
        block: None,
        wall: None,
        liquid: None,
        wiring: None,
        run_length: RunLength {
            length: 98,
            rle_type: SingleByte,
        },
        position: Position {
            x: 0,
            y: 0,
        },
    },
    Tile {
        tile_header: TileHeader {
            has_block: false,
            has_attributes: false,
            has_wall: false,
            liquid_type: NoLiquid,
            has_extended_block_id: false,
            rle_type: SingleByte,
            bits: TBitVec(
                BitVec<Lsb0, u8> [
                    0b00000010,
                ],
            ),
        },
        attributes: None,
        ext_attributes: None,
        block: None,
        wall: None,
        liquid: None,
        wiring: None,
        run_length: RunLength {
            length: 98,
            rle_type: SingleByte,
        },
        position: Position {
            x: 0,
            y: 0,
        },
    },
    Tile {
        tile_header: TileHeader {
            has_block: false,
            has_attributes: false,
            has_wall: false,
            liquid_type: NoLiquid,
            has_extended_block_id: false,
            rle_type: SingleByte,
            bits: TBitVec(
                BitVec<Lsb0, u8> [
                    0b00000010,
                ],
            ),
        },
        attributes: None,
        ext_attributes: None,
        block: None,
        wall: None,
        liquid: None,
        wiring: None,
        run_length: RunLength {
            length: 98,
            rle_type: SingleByte,
        },
        position: Position {
            x: 0,
            y: 0,
        },
    },
    Tile {
        tile_header: TileHeader {
            has_block: false,
            has_attributes: false,
            has_wall: false,
            liquid_type: NoLiquid,
            has_extended_block_id: false,
            rle_type: SingleByte,
            bits: TBitVec(
                BitVec<Lsb0, u8> [
                    0b00000010,
                ],
            ),
        },
        attributes: None,
        ext_attributes: None,
        block: None,
        wall: None,
        liquid: None,
        wiring: None,
        run_length: RunLength {
            length: 98,
            rle_type: SingleByte,
        },
        position: Position {
            x: 0,
            y: 0,
        },
    },
    Tile {
        tile_header: TileHeader {
            has_block: false,
            has_attributes: false,
            has_wall: false,
            liquid_type: NoLiquid,
            has_extended_block_id: false,
            rle_type: SingleByte,
            bits: TBitVec(
                BitVec<Lsb0, u8> [
                    0b00000010,
                ],
            ),
        },
        attributes: None,
        ext_attributes: None,
        block: None,
        wall: None,
        liquid: None,
        wiring: None,
        run_length: RunLength {
            length: 98,
            rle_type: SingleByte,
        },
        position: Position {
            x: 0,
            y: 0,
        },
    },
    Tile {
        tile_header: TileHeader {
            has_block: false,
            has_attributes: false,
            has_wall: false,
            liquid_type: NoLiquid,
            has_extended_block_id: false,
            rle_type: SingleByte,
            bits: TBitVec(
                BitVec<Lsb0, u8> [
                    0b00000010,
                ],
            ),
        },
        attributes: None,
        ext_attributes: None,
        block: None,
        wall: None,
        liquid: None,
        wiring: None,
        run_length: RunLength {
            length: 98,
            rle_type: SingleByte,
        },
        position: Position {
            x: 0,
            y: 0,
        },
    },
    Tile {
        tile_header: TileHeader {
            has_block: false,
            has_attributes: false,
            has_wall: false,
            liquid_type: NoLiquid,
            has_extended_block_id: false,
            rle_type: SingleByte,
            bits: TBitVec(
                BitVec<Lsb0, u8> [
                    0b00000010,
                ],
            ),
        },
        attributes: None,
        ext_attributes: None,
        block: None,
        wall: None,
        liquid: None,
        wiring: None,
        run_length: RunLength {
            length: 98,
            rle_type: SingleByte,
        },
        position: Position {
            x: 0,
            y: 0,
        },
    },
    Tile {
        tile_header: TileHeader {
            has_block: false,
            has_attributes: false,
            has_wall: false,
            liquid_type: NoLiquid,
            has_extended_block_id: false,
            rle_type: SingleByte,
            bits: TBitVec(
                BitVec<Lsb0, u8> [
                    0b00000010,
                ],
            ),
        },
        attributes: None,
        ext_attributes: None,
        block: None,
        wall: None,
        liquid: None,
        wiring: None,
        run_length: RunLength {
            length: 98,
            rle_type: SingleByte,
        },
        position: Position {
            x: 0,
            y: 0,
        },
    },
    Tile {
        tile_header: TileHeader {
            has_block: false,
            has_attributes: false,
            has_wall: false,
            liquid_type: NoLiquid,
            has_extended_block_id: false,
            rle_type: SingleByte,
            bits: TBitVec(
                BitVec<Lsb0, u8> [
                    0b00000010,
                ],
            ),
        },
        attributes: None,
        ext_attributes: None,
        block: None,
        wall: None,
        liquid: None,
        wiring: None,
        run_length: RunLength {
            length: 98,
            rle_type: SingleByte,
        },
        position: Position {
            x: 0,
            y: 0,
        },
    },
    Tile {
        tile_header: TileHeader {
            has_block: false,
            has_attributes: false,
            has_wall: false,
            liquid_type: NoLiquid,
            has_extended_block_id: false,
            rle_type: SingleByte,
            bits: TBitVec(
                BitVec<Lsb0, u8> [
                    0b00000010,
                ],
            ),
        },
        attributes: None,
        ext_attributes: None,
        block: None,
        wall: None,
        liquid: None,
        wiring: None,
        run_length: RunLength {
            length: 98,
            rle_type: SingleByte,
        },
        position: Position {
            x: 0,
            y: 0,
        },
    },
    Tile {
        tile_header: TileHeader {
            has_block: false,
            has_attributes: false,
            has_wall: false,
            liquid_type: NoLiquid,
            has_extended_block_id: false,
            rle_type: SingleByte,
            bits: TBitVec(
                BitVec<Lsb0, u8> [
                    0b00000010,
                ],
            ),
        },
        attributes: None,
        ext_attributes: None,
        block: None,
        wall: None,
        liquid: None,
        wiring: None,
        run_length: RunLength {
            length: 98,
            rle_type: SingleByte,
        },
        position: Position {
            x: 0,
            y: 0,
        },
    },
    Tile {
        tile_header: TileHeader {
            has_block: false,
            has_attributes: false,
            has_wall: false,
            liquid_type: NoLiquid,
            has_extended_block_id: false,
            rle_type: SingleByte,
            bits: TBitVec(
                BitVec<Lsb0, u8> [
                    0b00000010,
                ],
            ),
        },
        attributes: None,
        ext_attributes: None,
        block: None,
        wall: None,
        liquid: None,
        wiring: None,
        run_length: RunLength {
            length: 98,
            rle_type: SingleByte,
        },
        position: Position {
            x: 0,
            y: 0,
        },
    },
    Tile {
        tile_header: TileHeader {
            has_block: false,
            has_attributes: false,
            has_wall: false,
            liquid_type: NoLiquid,
            has_extended_block_id: false,
            rle_type: SingleByte,
            bits: TBitVec(
                BitVec<Lsb0, u8> [
                    0b00000010,
                ],
            ),
        },
        attributes: None,
        ext_attributes: None,
        block: None,
        wall: None,
        liquid: None,
        wiring: None,
        run_length: RunLength {
            length: 98,
            rle_type: SingleByte,
        },
        position: Position {
            x: 0,
            y: 0,
        },
    },
    Tile {
        tile_header: TileHeader {
            has_block: false,
            has_attributes: false,
            has_wall: false,
            liquid_type: NoLiquid,
            has_extended_block_id: false,
            rle_type: SingleByte,
            bits: TBitVec(
                BitVec<Lsb0, u8> [
                    0b00000010,
                ],
            ),
        },
        attributes: None,
        ext_attributes: None,
        block: None,
        wall: None,
        liquid: None,
        wiring: None,
        run_length: RunLength {
            length: 98,
            rle_type: SingleByte,
        },
        position: Position {
            x: 0,
            y: 0,
        },
    },
    Tile {
        tile_header: TileHeader {
            has_block: false,
            has_attributes: false,
            has_wall: false,
            liquid_type: NoLiquid,
            has_extended_block_id: false,
            rle_type: SingleByte,
            bits: TBitVec(
                BitVec<Lsb0, u8> [
                    0b00000010,
                ],
            ),
        },
        attributes: None,
        ext_attributes: None,
        block: None,
        wall: None,
        liquid: None,
        wiring: None,
        run_length: RunLength {
            length: 98,
            rle_type: SingleByte,
        },
        position: Position {
            x: 0,
            y: 0,
        },
    },
    Tile {
        tile_header: TileHeader {
            has_block: false,
            has_attributes: false,
            has_wall: false,
            liquid_type: NoLiquid,
            has_extended_block_id: false,
            rle_type: SingleByte,
            bits: TBitVec(
                BitVec<Lsb0, u8> [
                    0b00000010,
                ],
            ),
        },
        attributes: None,
        ext_attributes: None,
        block: None,
        wall: None,
        liquid: None,
        wiring: None,
        run_length: RunLength {
            length: 98,
            rle_type: SingleByte,
        },
        position: Position {
            x: 0,
            y: 0,
        },
    },
    Tile {
        tile_header: TileHeader {
            has_block: false,
            has_attributes: false,
            has_wall: false,
            liquid_type: NoLiquid,
            has_extended_block_id: false,
            rle_type: SingleByte,
            bits: TBitVec(
                BitVec<Lsb0, u8> [
                    0b00000010,
                ],
            ),
        },
        attributes: None,
        ext_attributes: None,
        block: None,
        wall: None,
        liquid: None,
        wiring: None,
        run_length: RunLength {
            length: 98,
            rle_type: SingleByte,
        },
        position: Position {
            x: 0,
            y: 0,
        },
    },
    Tile {
        tile_header: TileHeader {
            has_block: false,
            has_attributes: false,
            has_wall: false,
            liquid_type: NoLiquid,
            has_extended_block_id: false,
            rle_type: SingleByte,
            bits: TBitVec(
                BitVec<Lsb0, u8> [
                    0b00000010,
                ],
            ),
        },
        attributes: None,
        ext_attributes: None,
        block: None,
        wall: None,
        liquid: None,
        wiring: None,
        run_length: RunLength {
            length: 98,
            rle_type: SingleByte,
        },
        position: Position {
            x: 0,
            y: 0,
        },
    },
    Tile {
        tile_header: TileHeader {
            has_block: false,
            has_attributes: false,
            has_wall: false,
            liquid_type: NoLiquid,
            has_extended_block_id: false,
            rle_type: SingleByte,
            bits: TBitVec(
                BitVec<Lsb0, u8> [
                    0b00000010,
                ],
            ),
        },
        attributes: None,
        ext_attributes: None,
        block: None,
        wall: None,
        liquid: None,
        wiring: None,
        run_length: RunLength {
            length: 98,
            rle_type: SingleByte,
        },
        position: Position {
            x: 0,
            y: 0,
        },
    },
    Tile {
        tile_header: TileHeader {
            has_block: false,
            has_attributes: false,
            has_wall: false,
            liquid_type: NoLiquid,
            has_extended_block_id: false,
            rle_type: SingleByte,
            bits: TBitVec(
                BitVec<Lsb0, u8> [
                    0b00000010,
                ],
            ),
        },
        attributes: None,
        ext_attributes: None,
        block: None,
        wall: None,
        liquid: None,
        wiring: None,
        run_length: RunLength {
            length: 98,
            rle_type: SingleByte,
        },
        position: Position {
            x: 0,
            y: 0,
        },
    },
    Tile {
        tile_header: TileHeader {
            has_block: false,
            has_attributes: false,
            has_wall: false,
            liquid_type: NoLiquid,
            has_extended_block_id: false,
            rle_type: SingleByte,
            bits: TBitVec(
                BitVec<Lsb0, u8> [
                    0b00000010,
                ],
            ),
        },
        attributes: None,
        ext_attributes: None,
        block: None,
        wall: None,
        liquid: None,
        wiring: None,
        run_length: RunLength {
            length: 98,
            rle_type: SingleByte,
        },
        position: Position {
            x: 0,
            y: 0,
        },
    },
    Tile {
        tile_header: TileHeader {
            has_block: false,
            has_attributes: false,
            has_wall: false,
            liquid_type: NoLiquid,
            has_extended_block_id: false,
            rle_type: SingleByte,
            bits: TBitVec(
                BitVec<Lsb0, u8> [
                    0b00000010,
                ],
            ),
        },
        attributes: None,
        ext_attributes: None,
        block: None,
        wall: None,
        liquid: None,
        wiring: None,
        run_length: RunLength {
            length: 98,
            rle_type: SingleByte,
        },
        position: Position {
            x: 0,
            y: 0,
        },
    },
    Tile {
        tile_header: TileHeader {
            has_block: false,
            has_attributes: false,
            has_wall: false,
            liquid_type: NoLiquid,
            has_extended_block_id: false,
            rle_type: SingleByte,
            bits: TBitVec(
                BitVec<Lsb0, u8> [
                    0b00000010,
                ],
            ),
        },
        attributes: None,
        ext_attributes: None,
        block: None,
        wall: None,
        liquid: None,
        wiring: None,
        run_length: RunLength {
            length: 98,
            rle_type: SingleByte,
        },
        position: Position {
            x: 0,
            y: 0,
        },
    },
    Tile {
        tile_header: TileHeader {
            has_block: false,
            has_attributes: false,
            has_wall: false,
            liquid_type: NoLiquid,
            has_extended_block_id: false,
            rle_type: SingleByte,
            bits: TBitVec(
                BitVec<Lsb0, u8> [
                    0b00000010,
                ],
            ),
        },
        attributes: None,
        ext_attributes: None,
        block: None,
        wall: None,
        liquid: None,
        wiring: None,
        run_length: RunLength {
            length: 98,
            rle_type: SingleByte,
        },
        position: Position {
            x: 0,
            y: 0,
        },
    },
    Tile {
        tile_header: TileHeader {
            has_block: false,
            has_attributes: false,
            has_wall: false,
            liquid_type: NoLiquid,
            has_extended_block_id: false,
            rle_type: SingleByte,
            bits: TBitVec(
                BitVec<Lsb0, u8> [
                    0b00000010,
                ],
            ),
        },
        attributes: None,
        ext_attributes: None,
        block: None,
        wall: None,
        liquid: None,
        wiring: None,
        run_length: RunLength {
            length: 98,
            rle_type: SingleByte,
        },
        position: Position {
            x: 0,
            y: 0,
        },
    },
    Tile {
        tile_header: TileHeader {
            has_block: false,
            has_attributes: false,
            has_wall: false,
            liquid_type: NoLiquid,
            has_extended_block_id: false,
            rle_type: SingleByte,
            bits: TBitVec(
                BitVec<Lsb0, u8> [
                    0b00000010,
                ],
            ),
        },
        attributes: None,
        ext_attributes: None,
        block: None,
        wall: None,
        liquid: None,
        wiring: None,
        run_length: RunLength {
            length: 98,
            rle_type: SingleByte,
        },
        position: Position {
            x: 0,
            y: 0,
        },
    },
    Tile {
        tile_header: TileHeader {
            has_block: false,
            has_attributes: false,
            has_wall: false,
            liquid_type: NoLiquid,
            has_extended_block_id: false,
            rle_type: SingleByte,
            bits: TBitVec(
                BitVec<Lsb0, u8> [
                    0b00000010,
                ],
            ),
        },
        attributes: None,
        ext_attributes: None,
        block: None,
        wall: None,
        liquid: None,
        wiring: None,
        run_length: RunLength {
            length: 98,
            rle_type: SingleByte,
        },
        position: Position {
            x: 0,
            y: 0,
        },
    },
]
